src/main/java/org/apache/datasketches/GenericInequalitySearch.java

```diff
@@ -143,7 +143,7 @@ public static <T> int find(final T[] arr, final int low, final int high, final T
       if (hi - lo <= 1) {
         return resolve(arr, lo, hi, v, crit, comparator);
       }
-      final int mid = (lo + hi) / 2;
+      final int mid = lo + (hi - lo) / 2;
```

Comment:
not sure why this change to equivalent but slightly more complicated form

Reply:
The (lo + hi) / 2 especially with integers has a risk of overflow, while the lo + (hi - lo) / 2 does not.

Reply:
right, makes sense